diff --git a/dist/index.js b/dist/index.js
index 298d316e8c810bc43c32e01098dd22b93053c86a..090e8664a5751ba92165419809f28dd104560bcc 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -552,19 +552,7 @@ var urlValidator = import_shapeshift2.s.string.url({
   allowedProtocols: ["http:", "https:", "discord:"]
 }).setValidationEnabled(isValidationEnabled);
 function validateRequiredButtonParameters(style, label, emoji, customId, url) {
-  if (url && customId) {
-    throw new RangeError("URL and custom id are mutually exclusive");
-  }
-  if (!label && !emoji) {
-    throw new RangeError("Buttons must have a label and/or an emoji");
-  }
-  if (style === import_v10.ButtonStyle.Link) {
-    if (!url) {
-      throw new RangeError("Link buttons must have a url");
-    }
-  } else if (url) {
-    throw new RangeError("Non-link buttons cannot have a url");
-  }
+
 }
 __name(validateRequiredButtonParameters, "validateRequiredButtonParameters");
 
