diff --git a/typings/index.d.ts b/typings/index.d.ts
index 78432a371ea86b9a907bc2fa25ec63d3c9077154..52d687d09559ea53db21f49ca1ddb3a74d1e8375 100644
--- a/typings/index.d.ts
+++ b/typings/index.d.ts
@@ -1164,7 +1164,7 @@ export class CommandInteractionOptionResolver<Cached extends CacheType = CacheTy
   public getChannel<T extends ChannelType = ChannelType>(
     name: string,
     required: true,
-    channelTypes?: T[],
+    channelTypes?: Readonly<T[]>,
   ): Extract<
     NonNullable<CommandInteractionOption<Cached>['channel']>,
     {
@@ -1179,7 +1179,7 @@ export class CommandInteractionOptionResolver<Cached extends CacheType = CacheTy
   public getChannel<T extends ChannelType = ChannelType>(
     name: string,
     required?: boolean,
-    channelTypes?: T[],
+    channelTypes?: Readonly<T[]>,
   ): Extract<
     NonNullable<CommandInteractionOption<Cached>['channel']>,
     {
@@ -4326,13 +4326,13 @@ export type ApplicationCommandData =
 
 export interface ApplicationCommandChannelOptionData extends BaseApplicationCommandOptionsData {
   type: CommandOptionChannelResolvableType;
-  channelTypes?: ApplicationCommandOptionAllowedChannelTypes[];
-  channel_types?: ApplicationCommandOptionAllowedChannelTypes[];
+  channelTypes?: Readonly<ApplicationCommandOptionAllowedChannelTypes[]>;
+  channel_types?: Readonly<ApplicationCommandOptionAllowedChannelTypes[]>;
 }
 
 export interface ApplicationCommandChannelOption extends BaseApplicationCommandOptionsData {
   type: ApplicationCommandOptionType.Channel;
-  channelTypes?: ApplicationCommandOptionAllowedChannelTypes[];
+  channelTypes?: Readonly<ApplicationCommandOptionAllowedChannelTypes[]>;
 }
 
 export interface ApplicationCommandRoleOptionData extends BaseApplicationCommandOptionsData {